src/Scene/Scene.cpp

```diff
@@ -63,9 +63,15 @@ void Scene::DestroyObject(GameObject* gameObject)
                        {
                            if (pGameObject == gameObject || pGameObject->GetTransform()->IsChildOf(gameObject->GetTransform()))
```

Comment:
I think we need to pass `true` here as recursive child detection is defaulted to false now.